httputil/request.go

```diff
@@ -172,14 +179,25 @@ func SetAuthType(req *http.Request, res *http.Response) {
 }
 
 func GetAuthType(res *http.Response) string {
-	auth := res.Header.Get("Www-Authenticate")
-	if len(auth) < 1 {
-		auth = res.Header.Get("Lfs-Authenticate")
+
+	authType := GetAuthTypeFromHeader(res, WwwAuthenticateHeader)
+	if authType == BasicAuthType {
+		// Let us check Lfs-Authenticate header to see if server supports NTML auth.
+		authType = GetAuthTypeFromHeader(res, LfsAuthenticateHeader)
 	}
 
-	if strings.HasPrefix(strings.ToLower(auth), "ntlm") {
-		return "ntlm"
+	return authType
+}
+
+func GetAuthTypeFromHeader(res *http.Response, headerName string) string {
+	authHeaders := res.Header[headerName]
+	for i := range authHeaders {
+		auth := authHeaders[i]
+
+		if strings.HasPrefix(strings.ToLower(auth), NtlmAuthType) {
+			return NtlmAuthType
+		}
 	}
 
-	return "basic"
+	return BasicAuthType
```

Comment:
Great question. It looks to me that git-lfs currently only supports NTML and basic authentication.  Kerberos for example, is more secure than NTLM, which is more secure than basic auth, Since Kerberos is not supported in git-lfs, it does not make sense to recognize it. I must say that according to https://blogs.msdn.microsoft.com/chiranth/2014/04/17/setting-up-kerberos-authentication-for-a-website-in-iis/ when server sends Negotiate header, it indiacate that but both Kerberos and NTLM are supported. Since a lot of clients do not handle Negotiate, people normally configure servers to send both Negotiate and NTLM headers. Do you want to handle Negotiate explicitly?